src/app/auth/authentication.service.ts

```diff
@@ -23,6 +23,7 @@ export class AuthenticationService {
     if ('token' in params) {
       this.authToken = params['token'];
       localStorage.setItem('auth_token', this.authToken);
+      location.href = location.protocol + '//' + location.host;
```

Comment:
The token does not come in the URL but the page refresh is there.

We need to consider this scenario - the user needs to be sent back to the page from where he clicked login. 
If the user is on View Details and clicks Login, after the login is done, the View Details screen needs to be displayed. 
We need to store the route from where login was called, and after successful login, the user needs to be sent back to the same page.

@joshuawilson your thoughts?